cfn-resources/integration-tests/project/project_test.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package project
+
+import (
+	"bytes"
+	ctx "context"
+	"os"
+	"path"
+	"testing"
+	"text/template"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	cfn "github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	util "github.com/mongodb/mongodbatlas-cloudformation-resources/integration-tests/utility"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+type LocalTestContext struct {
+	cfnClient      *cfn.Client
+	atlasClient    *mongodbatlas.Client
+	projectTmplObj TestProject
+	resourceCtx    util.ResourceContext
+	template       string
+	err            error
+}
+
+type TestProject struct {
+	ResourceTypeName string
+	Name             string
+	OrgID            string
+	Profile          string
+	TeamID           string
+	ProjectName      string
+	ProjectID        string
+}
+
+const (
+	resourceTypeName  = "MongoDB::Atlas::Project"
+	resourceDirectory = "project"
+)
+
+var (
+	profile         = os.Getenv("ATLAS_SECRET_PROFILE")
+	orgID           = os.Getenv("ATLAS_ORG_ID")
+	e2eRandSuffix   = util.GetRandNum().String()
+	testProjectName = "cfn-e2e-project" + e2eRandSuffix
+	testTeamName    = "cfn-e2e-team" + e2eRandSuffix
+	stackName       = "stack-project-e2e-" + e2eRandSuffix
+)
+
+func TestProjectCFN(t *testing.T) {
+	testCtx := setupSuite(t)
+
+	t.Run("Validate Template", func(t *testing.T) {
+		testIsTemplateValid(t, testCtx)
+	})
+
+	t.Run("Create Stack", func(t *testing.T) {
+		testCreateStack(t, testCtx)
+	})
+
+	t.Run("Update Stack", func(t *testing.T) {
+		testUpdateStack(t, testCtx)
+	})
+
+	t.Run("Delete Stack", func(t *testing.T) {
+		testDeleteStack(t, testCtx)
+	})
+}
+
+func setupSuite(t *testing.T) *LocalTestContext {
+	t.Helper()
+	t.Log("Setting up suite")
+	testCtx := new(LocalTestContext)
+	testCtx.setUp(t)
+
+	return testCtx
+}
+
+func (c *LocalTestContext) setUp(t *testing.T) {
+	c.resourceCtx = util.InitResourceCtx(e2eRandSuffix, resourceTypeName, resourceDirectory)
+	c.setClients(t)
+	util.PublishToPrivateRegistry(t, c.resourceCtx)
+	c.setupPrerequisites(t)
+}
```

Comment:
Shouldn't this function be in the `atlas_helper` file so that it can be used by other tests?

Reply:
since prerequisites will be different for each resource, didn't add it to the helper